### ISSUE TITLE
Wrapper for UNUserNotificationCenter

### ISSUE THREAD

**Issue by Miiha** [pointfreeco/swift-composable-architecture#310]
Hi, I was working on a struct wrapper for UNUserNotificationCenter, which resulted in this [gist](https://gist.github.com/Miiha/5b26556e2d24ef190cef5fe383cba8c4). 
It currently uses vanilla Combine and not the CTA-Effect type. There are also some rough edges around the mapped notification related types.
Using your approach it would now be possible to test various aspects of the user notifications framework.
Could this be something you are interested in adding as an additional wrapper library?

**Comment by andreyz** [comment]
@Miiha would you like to convert that to a fork on your account and we could collaborate on polishing it? I'm about to adapt notifications center to TCA in my own app, so this perfectly aligns with my priorities.

**Comment by Miiha** [comment]
Hi @andreyz
if you could transfer it, that would be great 👍

**Comment by mbrandonw** [comment]
Hey @Miiha and @andreyz, thanks for picking this up! We have also written a small UserNotifications wrapper for a project we are working on, and someday hope to polish it up for public consumption (here's a [gist](https://gist.github.com/mbrandonw/02c0adee4e2ee5c4c72476e09769bacd)). Looks very similar to what you have done, though we haven't implemented all of the functionality yet.

We feel that right now it would be most appropriate to keep this library external to `ComposableArchitecture`. In fact, we plan on moving our core motion and core location wrappers out into their own repo too. I just explained why in this PR #313. Long story short, we plan on cooking up a "Composable Architecture Community" organization where we can house all of these wrappers and extension libraries. So maybe someday in the near future you will be able to transfer your repo over to that org if you are interested.

Cheers!

**Comment by Miiha** [comment]
Hi @mbrandonw, makes absolute sense to keep the wrapper libraries separate from the Composable Architecture repo. 
I took a look at your gist and incorporated some ideas into my repo.
The full UserNotifications API should now be implemented with support for all platforms. 
There where some quirks I had to work around, but I think it should now be working.
Let me know what you think, I am always open for input. 

https://github.com/Miiha/ComposableUserNotifications

I am of course happy moving it into a future Composable Architecture Community.
